### PR TITLE
Update constants.py to reflect consistent spelling

### PR DESCRIPTION
**Pull request by sudocurse**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
constants.py

##### SUMMARY
Updating to reflect the inconsistent spelling of 'behavior' throughout the constants file. Based on the suggestion given in https://github.com/ansible/ansible/issues/7091#issuecomment-40954397

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure due to PEP 8 issue:

```
2017-03-01 02:38:29 ERROR: PEP 8: lib/ansible/constants.py:226:29: E126 continuation line over-indented for hanging indent (legacy)
```

The PEP 8 tests can be run locally with `make pep8`.

File: lib/ansible/constants.py [hunk 1]
```diff
@@ -221,7 +221,8 @@ def load_config_file():
 DEFAULT_MANAGED_STR       = get_config(p, DEFAULTS, 'ansible_managed',  None,           'Ansible managed')
 DEFAULT_SYSLOG_FACILITY   = get_config(p, DEFAULTS, 'syslog_facility',  'ANSIBLE_SYSLOG_FACILITY', 'LOG_USER')
 DEFAULT_KEEP_REMOTE_FILES = get_config(p, DEFAULTS, 'keep_remote_files', 'ANSIBLE_KEEP_REMOTE_FILES', False, value_type='boolean')
-DEFAULT_HASH_BEHAVIOUR    = get_config(p, DEFAULTS, 'hash_behaviour', 'ANSIBLE_HASH_BEHAVIOUR', 'replace')
+DEFAULT_HASH_BEHAVIOUR    = get_config(p, DEFAULTS, 'hash_behaviour', 'ANSIBLE_HASH_BEHAVIOUR', 'replace') \
+                            or get_config(p, DEFAULTS, 'hash_behavior', 'ANSIBLE_HASH_BEHAVIOUR', 'replace')
```
Comment (abadger) [inline lib/ansible/constants.py:225]: environment variable should also have the alternate spelling.

**Comment by abadger** [comment]
We're okay with making an alias here but the implementation has some problems.

* It is not a general solution (for instance, booleans can't use this)
* It makes it so config vars can override environment variables (config var hash_behaviour=merge will override environment variable HASH_BEHAVIOR=replace)

I think the only way to do this is to modify get_config to take an alias definition.  Maybe allow key and env_var to be either sequences (list/tuple) or a string.  In _get_config(), detect if it's a string, make it a list.  Then iterate on the list of keys and list of env_vars to find the proper setting.

needs_revision

**Comment by sivel** [comment]
We somewhat did this when we deprecated `hostsfile`/`ANSIBLE_HOSTS`. 

https://github.com/ansible/ansible/blob/devel/lib/ansible/constants.py#L187
https://github.com/ansible/ansible/blob/devel/lib/ansible/constants.py#L199

I believe if we follow that pattern, it would work.

I remember some issues with that implementation, but I think it was due to how OS packagers shipped a default `/etc/ansible/ansible.cfg`.

**Comment by samdoran** [comment]
Thank you very much for your submission to Ansible.

Unfortunately, we're not sure if we want this feature in the program. Our reasons for this are:

   * `constants.py` has been completely refactored to work with the new `ansible-config` 
   * With `ansible-config`, it's actually very easy to do what you're suggesting

However, we're absolutely always up for discussion. Since this is a really busy project, we don't always see comments on closed tickets, but want to encourage open dialog. You can stop by the development list, and we'd be glad to talk about it - and we might even be persuaded otherwise!

   * https://groups.google.com/forum/#!forum/ansible-devel

In the future, sometimes starting a discussion on the development list prior to implementing a feature can make getting things included a little easier, but it's not always necessary.

Thank you once again for this and your interest in Ansible.

**Comment by bcoca** [comment]
Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us that you've taken time to contribute.

Unfortunately, we're not sure if we want this feature in the program, and I don't want this to seem confrontational. 

However, we're absolutely always up for discussion.  Since this is a really busy project, we don't always see comments on closed tickets, but want to encourage
open dialog.  You can stop by the development list, and we'd be glad to talk about it - and we might even be persuaded otherwise!

   * https://groups.google.com/forum/#!forum/ansible-devel

In the future, sometimes starting a discussion on the development list prior to implenting a feature can make getting things included a little easier, but it's not always necessary.

Thank you once again for this and your interest in Ansible!

**Pull request closed without merging**